engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/UpdateByBucketHelper.java

```diff
@@ -294,9 +294,17 @@ private int fillChunkWithNonNull(LongChunk<OrderedRowKeys> keysChunk, LongChunk<
     public void prepareForUpdate(final TableUpdate upstream, final boolean initialStep) {
         Assert.eqFalse(isDirty, "UpdateBy bucket was marked dirty before processing an update");
 
+        final boolean timestampsModified;
+
         // add all the SSA data
         if (timestampColumnName != null) {
             processUpdateForSsa(upstream);
+
+            // test whether any timestamps were modified
+            timestampsModified =
+                    upstream.modified().isNonempty() && upstream.modifiedColumnSet().containsAny(timestampColumnSet);
```

Comment:
Maybe it would be best to do this check once and also pass the result into `processUpdateForSsa`?